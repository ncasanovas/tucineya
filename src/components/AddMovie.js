import React, { useState } from "react";
import axios from "axios";

import { Modal, Button } from "react-bootstrap";
import { useHistory } from "react-router";

export const AddMovie = () => {
  const [show, setShow] = useState(false);

  const history = useHistory();

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <div className="mt-2">
      <Button variant={"primary"} onClick={handleShow}>
        Agregar Pelicula
      </Button>

      <Modal
        show={show}
        onHide={handleClose}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>Pelicula</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          ¿Está seguro de querer eliminar los usuarios seleccionados?
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary btn-secondary" onClick={handleClose}>
            Volver
          </Button>
          <Button variant="danger">Eliminar</Button>
        </Modal.Footer>
      </Modal>
    </div>
  );
};
