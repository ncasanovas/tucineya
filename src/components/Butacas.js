import React, { useEffect, useState, useContext } from "react";
import axios from "axios";
import { useHistory, HashRouter, Link } from "react-router-dom";

import { AuthContext } from "../auth/AuthContext";
import { types } from "../types/types";
import { MovieContext } from "./MovieContext";

export const Butacas = () => {
  const asientosFila = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"];
  const filas = ["1", "2", "3", "4", "5", "6", "7"];
  const [butacaElegida, setButacaElegida] = useState([]);
  const [butacasConfirmadas, setButacasConfirmadas] = useState([]);
  const [confirmar, setConfirmar] = useState(false);
  const [precioSala, setPrecioSala] = useState(0);
  const [precioTotalSala, setPrecioTotalSala] = useState(0);
  const { dispatch } = useContext(AuthContext);
  const history = useHistory();
  const { movies, idSala, butacas } = useContext(MovieContext);

  useEffect(async () => {
    if (idSala !== null) {
      //setButacasConfirmadas(butacas.split(","));
       await axios
        //.get(`http://localhost:4000/api/butacas/${idSala}`)
        .get(`https://tucineya.herokuapp.com/api/butacas/${idSala}`)
        .then((res) => {
          console.log(res);
          if (res.data !== null) {
            if (
              res.data[0][0].butacas === "" ||
              res.data[0][0].butacas === null
            ) {
              localStorage.setItem("butacas", "");
              setButacasConfirmadas(localStorage.getItem("butacas"));
            } else {
              let butacas = res.data[0][0].butacas.split(",");
              localStorage.setItem("butacas", butacas);
              setButacasConfirmadas(butacas);
            }
          } else {
            setButacasConfirmadas(localStorage.getItem("butacas"));
          }
        }); 
      await axios
        //.post(`http://localhost:4000/api/sala/${idSala}`)
        .post(`https://tucineya.herokuapp.com/api/sala/${idSala}`)
        .then((res) => {
          console.log(res.data.precio);
          setPrecioSala(res.data.precio); //Está peticion me trae el precio de la sala
        });
    }
  }, [confirmar]);

  const seleccionarButaca = (fila, asiento) => {
    if (butacaElegida.includes(fila + asiento)) {
      // includes busca si el asiento ya está en el array "butacaElegida" devuelve true
      let index = butacaElegida.indexOf(fila + asiento); // Guardo el indice en donde esté la butaca
      butacaElegida.splice(index, 1); // Y la borro del array segun el indice que encontró, el segundo argumento es para que borre solo ese registro, que borre 1 solo indice apenas en cuentre coincidencias
      setButacaElegida([...butacaElegida]); // setea las butacas con el array y ya sin la que borramos
      setPrecioTotalSala(precioTotalSala - precioSala); //Seteo el precio total restando el total y el precio de sala cuando deseleccione una butaca disponible
    } else {
      setButacaElegida([...butacaElegida, fila + asiento]); // setea las butacas seleccionadas si no está en el array de butacaElegida
      setPrecioTotalSala(precioTotalSala + precioSala); //Seteo el precio total sumando el total y el precio de sala cuando seleccione una butaca disponible
    }
  };

  const confirmarSeleccion = async () => {
    if (butacaElegida.length !== 0) {
      await axios
        .post("https://tucineya.herokuapp.com/api/butacas", {
        //.post("http://localhost:4000/api/butacas", {
          butacas: butacaElegida,
          idSala: idSala,
        })
        .then(() => {
          setButacaElegida([]);
          setConfirmar(!confirmar);
          setPrecioTotalSala(0); //Seteo el precio total a 0 para que no siga sumando cuando seleccione más butacas
        });
    }
  };

  const handleLogout = () => {
    dispatch({
      type: types.logout,
    });
    history.replace("./");
  };

  return (
    <div className="row">
      <div className="d-flex justify-content-end">
        <HashRouter>
          <Link
            className="col-2 align-items-center"
            onClick={() => {
              localStorage.removeItem("butacas");
            }}
            to="/buscarPelicula"
          >
            Atras
          </Link>
        </HashRouter>
        <button
          className="nav-item nav-link btn white-text"
          onClick={handleLogout}
        >
          Cerrar Sesión
        </button>
      </div>
      {movies === [] ? null : (
        <div className="col">
          <h1 className="text-white">{movies[0].idNombrePelicula}</h1>
          <img
            src={movies[0].posterPelicula}
            alt={movies[0].idNombrePelicula}
          />
        </div>
      )}
      <div className="col">
        <ul className="showcase">
          <li>
            <div className="seat"></div>
            <small>Disponible</small>
          </li>
          <li>
            <div className="seat selected"></div>
            <small>Seleccionado</small>
          </li>
          <li>
            <div className="seat occupied"></div>
            <small>Ocupado</small>
          </li>
        </ul>

        <div className="container ">
          <div className="screen"></div>
          <div className="row">
            {filas.map((fila, j) => {
              return (
                <div className="row" key={j} value={fila}>
                  {asientosFila.map((asiento, i) => {
                    return (
                      <div
                        className={
                          butacasConfirmadas.includes(fila + asiento)
                            ? "seat occupied"
                            : butacaElegida.includes(fila + asiento)
                            ? "seat selected"
                            : "seat"
                        }
                        style={
                          butacasConfirmadas.includes(fila + asiento)
                            ? { pointerEvents: "none" }
                            : null
                        }
                        disabled
                        unselectable="true"
                        key={i}
                        value={asiento}
                        onClick={() => seleccionarButaca(fila, asiento)}
                      ></div>
                    );
                  })}
                </div>
              );
            })}
          </div>
          <button
            className="btn btn-primary"
            onClick={() => confirmarSeleccion()}
          >
            Confirmar
          </button>
        </div>

        <p className="text">
          Seleccionaste <span id="count">{butacaElegida.length}</span> asientos
          por el precio total de $<span id="total">{precioTotalSala}</span>!
        </p>
      </div>
    </div>
  );
};

