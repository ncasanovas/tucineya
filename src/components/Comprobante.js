import React from "react";
import { useHistory, HashRouter, Link } from "react-router-dom";

export const Comprobante = () => {
  const printReceipt = () => {
    var prtContent = document.getElementById("tarjeta");
    var WinPrint = window.open(
      "",
      "",
      "left=200,top=200,width=800,height=900,toolbar=0,scrollbars=0,status=0"
    );
    WinPrint.document.write(prtContent.innerHTML);
    WinPrint.document.close();
    WinPrint.focus();
    WinPrint.print();
    WinPrint.close();
  };

  const printReceipt2 = () => {
    var prtContent = document.getElementById("tarjeta2");
    var WinPrint = window.open(
      "",
      "",
      "left=0,top=0,width=800,height=900,toolbar=0,scrollbars=0,status=0"
    );
    WinPrint.document.write(prtContent.innerHTML);
    WinPrint.document.close();
    WinPrint.focus();
    WinPrint.print();
    WinPrint.close();
  };

  const { idNombrePelicula, idNumeroSala, fechaYhora, nombre } = JSON.parse(
    sessionStorage.getItem("Pelicula")
  );

  return (
    <div className="App">
      <header className="App-header">
        <div className="card">
          <HashRouter>
            <Link className="col-1" to="/buscarPelicula">
              Atras
            </Link>
          </HashRouter>
          <div className="card-body" id="tarjeta2" z-index="3200">
            <h5 className="card-title">Comprobante</h5>

            <div>
              <h3>{idNombrePelicula[0]}</h3>
              <p>Sala numero:{idNumeroSala}</p>
              <p>Precio: {sessionStorage.getItem("precioTotal")}</p>
              <p>Cine: {nombre}</p>
              <p>
                Butacas elegidas:{" "}
                {JSON.stringify(sessionStorage.getItem("butacasElegidas"))}
              </p>
              <p>Fecha: {fechaYhora}</p>

              <h1>Tu Cine Ya</h1>
            </div>
          </div>
        </div>

        <button
          className="hide-on-print btn btn-primary"
          onClick={printReceipt2}
        >
          Imprimir
        </button>
      </header>
    </div>
  );
};

export default Comprobante;
