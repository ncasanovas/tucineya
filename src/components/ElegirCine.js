import React, { useContext, useState } from "react";
import { Dropdown } from "react-bootstrap";
import axios from "axios";
import { MovieContext } from "./MovieContext";

export const ElegirCine = ({ localidades }) => {
  const [barrio, setBarrio] = useState([]);
  const [over, setOver] = useState();

  const { setMovies, setIdCine } = useContext(MovieContext);

  const onClick = async (localidad) => {
    await axios
      //.get(`http://localhost:4000/api/cines/${localidad.idLocalidad}`)
      .get(`https://tucineya.herokuapp.com/api/cines/${localidad.idLocalidad}`)
      .then((res) => {
        setBarrio(res.data[0]);
      });
  };

  const verPeliculas = async (barrio) => {
    if (barrio !== null) {
      await axios
        /* .get(`http://localhost:4000/api/cines/peliculas/${barrio.idCine}`) */
        .get(
          `https://tucineya.herokuapp.com/api/cines/peliculas/${barrio.idCine}`
        )
        .then((res) => {
          setMovies(res.data[0]);
          setIdCine(barrio.idCine);
        });
    }
  };

  return (
    <Dropdown>
      <Dropdown.Toggle variant="primary" id="dropdown-basic">
        {barrio.length === 0 ? "Localidades" : barrio[0].Localidad}
      </Dropdown.Toggle>

      <Dropdown.Menu>
        {localidades === undefined
          ? null
          : localidades.map((localidad, i) => {
              return (
                <Dropdown.Item key={i} onClick={() => onClick(localidad)}>
                  {localidad.nombreLocalidad}
                </Dropdown.Item>
              );
            })}
      </Dropdown.Menu>
      <div>
        <ul className="list-group list-group-flush">
          {barrio.length === 0
            ? null
            : barrio.map((barrio, i) => {
                return (
                  <div key={i} className="d-flex justify-content-center">
                    <li
                      className={
                        over === i
                          ? "list-group-item active"
                          : "list-group-item"
                      }
                      onClick={() => verPeliculas(barrio)}
                      style={{ cursor: "pointer" }}
                      value={i}
                      onMouseOver={(e) => setOver(e.target.value)}
                      onMouseOut={(e) => setOver()}
                    >
                      {barrio.Barrio}
                      {" - "}
                      {barrio.nombre} {" , "} {barrio.ubicacion}
                    </li>
                  </div>
                );
              })}
        </ul>
      </div>
    </Dropdown>
  );
};
