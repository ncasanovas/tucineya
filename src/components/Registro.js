import React, { useState } from "react";
import { HashRouter, Link } from "react-router-dom";
import axios from "axios";

export const Registro = () => {
  const [user, setUser] = useState({
    nombre: "",
    apellido: "",
    email: "",
    username: "",
    password: "",
    rePassword: "",
    celular: "",
    direccion: "",
    piso: "",
    dpto: "",
  });

  const onChange = (e) => {
    setUser({
      ...user,
      [e.target.name]: e.target.value,
      
    });
    console.log(user.nombre);
  };

  const validar = (user) => {
    const errors = {};

    if (user.nombre.length > 15) {
      errors.nombre = "Debe contener 15 letras o menos";
    }
    
    if(!/[A-Za-z]/.test(user.nombre)) {
      errors.nombre = "No debe contener caracteres especiales ni números";
    }

    if (user.apellido.length > 20) {
      errors.apellido = "Debe contener 20 letras o menos";
    }

    if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(user.email)) {
      errors.email = "Email invalido";
    }

    if (user.username.length > 10) {
      errors.username = "Debe contener 10 letras o menos";
    }

    if (!user.password) {
      errors.password = "Requerido";
    }

    if (!user.repassword) {
      errors.repassword = "Requerido";
    } else if (user.repassword !== user.password) {
      errors.repassword = "Las contraseñas deben coincidir";
    }

    if (
      parseInt(!user.celular.match(/^[0-9]{8,12}$/i)) ||
      parseInt(user.celular.match(/^[0-9]{12}$/i))
    ) {
      errors.celular = "Celular invalido";
    }

    if (!user.direccion) {
      errors.direccion = "Requerido";
    }

    if (!/^[0-9]{8,12}$/i) {
      errors.piso = "El piso debe ser numerico";
    }

    if (!/^[A-Z0-9._%+-]{1,2}/) {
      errors.dpto = "Departamento invalido";
    }

    return errors;
  };

  const onSubmit = async (e) => {
    e.preventDefault();
    e.target.reset();

    if (Object.keys(validar(user)).length === 0) {
      await axios
        .post("https://tucineya.herokuapp.com/api/register/", user)
        //.post("http://localhost:4000/api/register/", user)
        .then((res) => {
          alert(res.data.message);
          console.log(res);
        })
        .catch((err) => {
          console.log(err);
        });
    } else {
      alert(JSON.stringify(validar(user)));
    }

    setUser({
      nombre: "",
      apellido: "",
      email: "",
      username: "",
      password: "",
      rePassword: "",
      celular: "",
      direccion: "",
      piso: "",
      dpto: "",
    });
  };

  return (
    <>
      <div className="mb-3 row">
        <div className="justify-content-center">
          <div className="row">
            <HashRouter>
              <Link to="/" className="col-2 align-items-center">
                Atras
              </Link>
            </HashRouter>
            <p className="fs-1 col-8">FORMULARIO DE REGISTRO</p>
          </div>
          <form onSubmit={onSubmit} className="px-4 py-3">
            <p>DATOS PERSONALES</p>
            <div className="form-group row">
              <div className="col">
                <input
                  type="text"
                  id="nombre"
                  className="form-control mb-3"
                  name="nombre"
                  placeholder="Nombre"
                  onChange={onChange}
                  required
                />
                <input
                  type="text"
                  id="apellido"
                  className="form-control mb-3"
                  name="apellido"
                  placeholder="Apellido"
                  onChange={onChange}
                  required
                />
                <input
                  type="email"
                  id="email"
                  className="form-control mb-3"
                  name="email"
                  placeholder="Mail"
                  onChange={onChange}
                  required
                />
                <input
                  type="text"
                  id="username"
                  className="form-control mb-3"
                  name="username"
                  placeholder="Nombre de usuario"
                  onChange={onChange}
                  required
                />
                <input
                  type="text"
                  id="lname"
                  className="form-control mb-3"
                  name="password"
                  placeholder="Contraseña"
                  onChange={onChange}
                  required
                />
                <input
                  type="text"
                  id="lname"
                  className="form-control mb-3"
                  name="repassword"
                  placeholder="Confirma Contraseña"
                  onChange={onChange}
                  required
                />
              </div>
              <div className="col">
                <input
                  type="number"
                  id="celular"
                  className="form-control mb-3"
                  name="celular"
                  placeholder="Celular"
                  onChange={onChange}
                  required
                />
                <input
                  type="text"
                  id="direccion"
                  className="form-control mb-3"
                  name="direccion"
                  placeholder="Dirección"
                  onChange={onChange}
                  required
                />
                <input
                  type="number"
                  id="piso"
                  className="form-control mb-3"
                  name="piso"
                  placeholder="Piso"
                  onChange={onChange}
                />
                <input
                  type="text"
                  id="dpto"
                  className="form-control mb-3"
                  name="dpto"
                  placeholder="Dpto"
                  onChange={onChange}
                />
                <button type="submit" className="btn btn-success mb-3">
                  Guardar
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </>
  );
};
