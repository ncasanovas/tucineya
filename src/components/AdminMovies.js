import React, { useContext } from "react";
import { Card, Button } from "react-bootstrap";
import { AddMovie } from "./AddMovie";
import { MoviesContext } from "./MoviesContext";

export const AdminMovies = () => {
  const { moviesContext } = useContext(MoviesContext);

  return (
    <div className="p-0 m-0">
      <div>
        {moviesContext ? (
          <div className="overflow-auto d-flex">
            {moviesContext.map((movie, i) => {
              return (
                <div className="col-4 overflow-auto d-flex" key={i}>
                  <Card>
                    <Card.Img
                      className="img-fluid"
                      id="imgPelicula"
                      variant="top"
                      src={`https://${movie.posterPelicula}`}
                    />
                    <Card.Body>
                      <Card.Title id="cardText">
                        {movie.nombrePelicula}
                      </Card.Title>
                      <Card.Text>{movie.sinopsis}</Card.Text>
                    </Card.Body>
                  </Card>
                </div>
              );
            })}
          </div>
        ) : (
          <div className="d-flex justify-content-center">
            <div
              className="spinner-border text-primary d-flex justify-content-center"
              role="status"
            >
              <span className="visually-hidden ">Loading...</span>
            </div>
          </div>
        )}
      </div>
      {moviesContext ? (
        <AddMovie />
      ) : //<Button variant="primary">Agregar Pelicula</Button>
      null}
    </div>
  );
};
