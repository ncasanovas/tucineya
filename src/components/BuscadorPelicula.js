import React, { useState, useContext, useEffect } from "react";
import axios from "axios";
import { useHistory } from "react-router-dom";

import { AuthContext } from "../auth/AuthContext";
import { MovieContext } from "./MovieContext";

import { types } from "../types/types";
import { ResultadoPeliculas } from "./ResultadoPeliculas";
import { ElegirCine } from "./ElegirCine";

export const BuscadorPelicula = () => {
  const [inputValue, setInputValue] = useState();
  const [user, setUser] = useState();
  const [localidades, setLocalidades] = useState();
  const { dispatch } = useContext(AuthContext);
  const { movies, setMovies } = useContext(MovieContext);
  const history = useHistory();

  const onChangeinput = (e) => {
    setInputValue(e.target.value);
  };

  useEffect(async () => {
    let user = localStorage.getItem("usuario");
    setUser(user);
    await axios
      //.get("http://localhost:4000/api/cines/")
      .get("https://tucineya.herokuapp.com/api/cines/")
      .then((res) => {
        setLocalidades(res.data[0]);
      });
  }, [movies]);

  const onClickSearchMovie = async (e) => {
    e.target.reset();
    e.preventDefault();
    await axios
      //.get(`http://localhost:4000/api/movies/${inputValue}`)
      .get(`https://tucineya.herokuapp.com/api/movies/${inputValue}`)
      .then((res) => {
        setMovies(res.data.data[0]);
      })
      .catch((e) => {
        console.log(e);
      });
    setInputValue("");
  };

  const handleLogout = () => {
    console.log("Click");
    dispatch({
      type: types.logout,
    });
    history.replace("./");
  };

  return (
    <div className="row mt-4">
      <div className="d-flex justify-content-end">
        <h4 style={{ color: "white" }}>Bienvenido {user}!</h4>
        <button
          className="nav-item nav-link btn white-text"
          onClick={() => handleLogout}
        >
          Cerrar Sesi√≥n
        </button>
      </div>
      <div className="col mb-3">
        <form className="mb-4" onSubmit={() => onClickSearchMovie()}>
          <input
            type="text"
            placeholder="Busque por titulo"
            onChange={onChangeinput}
          />
          <button className="btn-primary">Buscar</button>
        </form>
        <div id="IdDropdown">
          <ElegirCine localidades={localidades} />
        </div>
      </div>
      <div className="col">{movies ? <ResultadoPeliculas /> : null}</div>
    </div>
  );
};
