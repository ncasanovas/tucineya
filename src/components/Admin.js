import React, { useState, useEffect, useContext } from "react";
import axios from "axios";
import { HashRouter, Link } from "react-router-dom";

import { AdminModal } from "./AdminModal";
import { AdminMovies } from "./AdminMovies";
import { MoviesContext } from "./MoviesContext";

export const Admin = () => {
  const [users, setUsers] = useState();
  const [deleteUsers, setDeleteUsers] = useState([]);

  const { setMoviesContext } = useContext(MoviesContext);

  useEffect(() => {
    const data = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/users/")
        .get("http://localhost:4000/api/users/")
        .then((res) => {
          setUsers(res.data[0]);
        });
    };
    const datamovies = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/movies/")
        .get("http://localhost:4000/api/movies/")
        .then((res) => {
          setMoviesContext(res.data[0]);
        });
    };
    data();
    datamovies();
  }, []);

  const onChangeCheck = (e) => {
    if (e.target.checked) {
      setDeleteUsers([...deleteUsers, e.target.value]);
    } else {
      for (let i = 0; i < deleteUsers.length; i++) {
        if (deleteUsers[i] === e.target.value) {
          deleteUsers.splice(i, 1);
          setDeleteUsers([...deleteUsers]);
        }
      }
    }
  };

  return (
    <div id="admin" className="container-fluid row">
      <div className="d-flex justify-content-end">
        <HashRouter>
          <Link to="/" className="col-2">
            Atras
          </Link>
        </HashRouter>
      </div>
      <div className="col-3 col-sm-4 col-md-3 col-lg-3">
        <h4 className="mb-4">Lista de usuarios</h4>
        <div className="row">
          {users ? (
            <ul className="list-group text-start col">
              {users.map((user, i) => {
                return (
                  <li className="list-group-item" key={i}>
                    <input
                      className="form-check-input me-1"
                      type="checkbox"
                      value={user.mail}
                      aria-label="..."
                      onChange={onChangeCheck}
                    />
                    {user.nombre} {user.apellido}
                  </li>
                );
              })}
            </ul>
          ) : (
            <div className="spinner-border text-primary m-6" role="status">
              <span className="visually-hidden ">Loading...</span>
            </div>
          )}
        </div>
        <div>
          {users ? (
            <div>
              <AdminModal
                deleteUsers={deleteUsers}
                setDeleteUsers={setDeleteUsers}
              />
            </div>
          ) : null}
        </div>
      </div>
      <div className="col-sm-12 col-md-9 col-lg-9">
        <h4 className="mb-4">Peliculas</h4>
        <AdminMovies />
      </div>
    </div>
  );
};

export default Admin;
