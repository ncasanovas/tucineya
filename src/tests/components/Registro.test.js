import React from "react";
import {shallow} from 'enzyme';
import {Registro} from "../../components/Registro";


describe('Pruebas en <Registro />', () => {

    const wrapper = shallow(<Registro />);
   

    test('debe de mostrarse correctamente', () => {
  
        expect(wrapper).toMatchSnapshot();

    });

  test('debe de tener un párrafo con el título FORMULARIO DE REGISTRO', () => {

      const title = 'FORMULARIO DE REGISTRO';

      const p = wrapper.find('p').at(0);
      expect(p.text().trim()).toBe(title);
  });

  test('debe de tener un párrafo que diga DATOS PERSONALES', () => {

    const title = 'DATOS PERSONALES';

    const p = wrapper.find('p').last();
 
    expect(p.text().trim()).toBe(title);
  });

  
  test('debe de tener los primeros 6 inputs', () => {

    const col = wrapper.find('.form-group').childAt(0);
 
    expect(col.children().length).toEqual(6);
  });

  test('debe de tener los 4 inputs restantes y el button', () => {

    const col = wrapper.find('.form-group').childAt(1);
    const button = wrapper.find('button');

    expect(col.children().length).toEqual(5);
    expect(col.children().last()).toEqual(button);
  });
});