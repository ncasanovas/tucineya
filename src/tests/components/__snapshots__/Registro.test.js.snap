// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Registro /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <div
    className="mb-3 row"
  >
    <div
      className="justify-content-center"
    >
      <div
        className="row"
      >
        <HashRouter>
          <Link
            className="col-2 align-items-center"
            to="/"
          >
            Atras
          </Link>
        </HashRouter>
        <p
          className="fs-1 col-8"
        >
          FORMULARIO DE REGISTRO
        </p>
      </div>
      <form
        className="px-4 py-3"
        onSubmit={[Function]}
      >
        <p>
          DATOS PERSONALES
        </p>
        <div
          className="form-group row"
        >
          <div
            className="col"
          >
            <input
              className="form-control mb-3"
              id="nombre"
              name="nombre"
              onChange={[Function]}
              placeholder="Nombre"
              required={true}
              type="text"
            />
            <input
              className="form-control mb-3"
              id="apellido"
              name="apellido"
              onChange={[Function]}
              placeholder="Apellido"
              required={true}
              type="text"
            />
            <input
              className="form-control mb-3"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Mail"
              required={true}
              type="email"
            />
            <input
              className="form-control mb-3"
              id="username"
              name="username"
              onChange={[Function]}
              placeholder="Nombre de usuario"
              required={true}
              type="text"
            />
            <input
              className="form-control mb-3"
              id="lname"
              name="password"
              onChange={[Function]}
              placeholder="Contraseña"
              required={true}
              type="text"
            />
            <input
              className="form-control mb-3"
              id="lname"
              name="repassword"
              onChange={[Function]}
              placeholder="Confirma Contraseña"
              required={true}
              type="text"
            />
          </div>
          <div
            className="col"
          >
            <input
              className="form-control mb-3"
              id="celular"
              name="celular"
              onChange={[Function]}
              placeholder="Celular"
              required={true}
              type="number"
            />
            <input
              className="form-control mb-3"
              id="direccion"
              name="direccion"
              onChange={[Function]}
              placeholder="Dirección"
              required={true}
              type="text"
            />
            <input
              className="form-control mb-3"
              id="piso"
              name="piso"
              onChange={[Function]}
              placeholder="Piso"
              type="number"
            />
            <input
              className="form-control mb-3"
              id="dpto"
              name="dpto"
              onChange={[Function]}
              placeholder="Dpto"
              type="text"
            />
            <button
              className="btn btn-success mb-3"
              type="submit"
            >
              Guardar
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</Fragment>
`;
